tests/test/org/commcare/backend/session/test/StackRegressionTests.java

```diff
@@ -0,0 +1,37 @@
+package org.commcare.backend.session.test;
+
+import org.commcare.core.interfaces.UserSandbox;
+import org.commcare.modern.session.SessionWrapper;
+import org.commcare.session.SessionDescriptorUtil;
+import org.commcare.session.SessionFrame;
+import org.commcare.test.utilities.MockApp;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by willpride on 9/15/16.
+ */
+
+public class StackRegressionTests {
+    /**
+     * Load form title from session where the case id is computed, not
+     * selected, and the case name is loaded from detail referenced by m0-f0,
+     * which is implicitly referenced
+     */
+    @Test
+    public void testGoBackAfterEndOfFormNavigation() throws Exception {
```

Comment:
This is the state that the given form would end up in after end of form navigation, so we want to test that in this situation it would behave correctly (IE determine that the case_id value would have been added by the case selection screen and properly resolve the unknown datum as such).